cmd/addon-operator-manager/main.go

```diff
@@ -71,10 +73,12 @@ func initReconcilers(mgr ctrl.Manager) {
 		os.Exit(1)
 	}
 
+	recorder := metrics.NewRecorder()
```

Comment:
let's bring the code to attach `recorder` to the `AddonReconciler` object under `SetupWithManager`?

Not that it would create much of a logical difference but from my perspective, that will keep the main.go clean and would also give the freedom to privatise the `Recorder` attribute in AddonReconciler struct to `AddonReconciler.recorder`


```go
(r *AddonReconciler) SetupWithManager() ... {
    r.recorder = metrics.NewRecorder()
}
```
wdyt :)

Reply:
We need to use a common instance of the recorder across all Reconcilers. For example, the same recorder is required in `AddonReconclier` as well: https://github.com/openshift/addon-operator/blob/e6dae36b8cd2f307829df2dd0a9b58c06274b3ba/cmd/addon-operator-manager/main.go#L96

If we move the declaration within `SetupWithManager`, we cannot initialize the Reconcilers with a common recorder.